probe-rs/src/architecture/arm/memory/adi_v5_memory_interface.rs

```diff
@@ -392,32 +462,98 @@ where
         Ok(())
     }
 
+    /// Read a block of 8 bit words at `address`.
+    ///
+    /// The number of words read is `data.len()`.
     pub fn read_8(
         &mut self,
         access_port: MemoryAp,
         address: u64,
         data: &mut [u8],
     ) -> Result<(), AccessPortError> {
+        if self.supports_only_32bit_data_size {
+            return Err(AccessPortError::UnsupportedTransferWidth(8));
+        }
+
         if data.is_empty() {
             return Ok(());
         }
 
-        let aligned = aligned_range(address, data.len())?;
-        let aligned_len = (aligned.end - aligned.start) as usize;
+        let start_address = address;
+        let mut data_u32 = vec![0u32; data.len()];
+
+        // Second we read in 32 bit reads until we have less than 32 bits left to read.
+        let csw = self.build_csw_register(DataSize::U8);
+        self.write_csw_register(access_port, csw)?;
+
+        let mut address = address;
+        self.write_tar_register(access_port, address)?;
+
+        // The maximum chunk size we can read before data overflows.
+        let max_chunk_size_bytes = 0x400;
+
+        let mut remaining_data_len = data.len();
+
+        let first_chunk_size_bytes = std::cmp::min(
+            max_chunk_size_bytes - (address as usize % max_chunk_size_bytes),
+            data.len(),
+        );
+
+        let mut data_offset = 0;
+
+        log::debug!(
+            "Read first block with len {} at address {:#08x}",
+            first_chunk_size_bytes,
+            address
+        );
+
+        let first_chunk_size_words = first_chunk_size_bytes;
```

Comment:
This makes sense in the context that this function reads 8bit words independently of what a word is on the target platform :)

Reply:
But there is no 8 bit word on the Cortex-M processors, a word is by definition 32 bit there. So please use another name to avoid confusion. (See https://en.wikipedia.org/wiki/Word_(computer_architecture) for reference.)

Reply:
Well that code wasn't introduced here and I still think it makes sense but I can try and think of something better.